hibernate-core/src/main/java/org/hibernate/boot/model/internal/OneToOneSecondPass.java

```diff
@@ -208,7 +208,7 @@ private void bindTargetManyToOne(
 		final KeyValue targetEntityIdentifier = targetEntity.getIdentifier();
 		boolean referenceToPrimaryKey = mappedBy == null
 				|| targetEntityIdentifier instanceof Component
-						&& !( (Component) targetEntityIdentifier ).hasProperty( mappedBy );
+						&& ( (Component) targetEntityIdentifier ).hasProperty( mappedBy );
```

Comment:
This was to solve the first error encountered with the user's reproducer. Without this, the mapping of the `IdClass` fails because it considers the wrong columns. Logically, it looked like `referenceToPrimaryKey` should be `true` if the `mappedBy` property is contained in the `targetEntityIdentifier`.

Reply:
@beikov the condition has been changed in 6.2 after a code refactoring

Reply:
Ok, it seems https://github.com/hibernate/hibernate-orm/commit/e87ad39dd6376c131bfa9574996e38cf1ff742c0 changed the logic here in a wrong way. So I guess the change is fine.